src/ray/rpc/grpc_server.cc

```diff
@@ -135,17 +135,22 @@ void GrpcServer::Run() {
 
   // Create calls for all the server call factories.
   for (auto &entry : server_call_factories_) {
-    for (int i = 0; i < num_threads_; i++) {
-      // Create a buffer of 100 calls for each RPC handler.
-      // TODO(edoakes): a small buffer should be fine and seems to have better
-      // performance, but we don't currently handle backpressure on the client.
-      int buffer_size = 100;
-      if (entry->GetMaxActiveRPCs() != -1) {
-        buffer_size = entry->GetMaxActiveRPCs();
-      }
-      for (int j = 0; j < std::max(1, buffer_size / num_threads_); j++) {
-        entry->CreateCall();
-      }
```

Comment:
What's wrong with original formula? it has ` / num_threads_`

Reply:
It's not the formula -- it's the nested loop:

 - `server_call_factories_` has already 1 object per thread
 - Then for every object we create at least `num_threads_` objects
 - This negates the setting of max active (inflight) RPCs (which is set globally)

Reply:
To summarize:

Currently each rpc handler can have at most `RAY_gcs_server_rpc_server_thread_num`* `RAY_gcs_max_active_rpcs_per_handler` active PRC calls. This PR changes so that each rpc handler has at most `RAY_gcs_max_active_rpcs_per_handler` active PRC calls.

This is a performance sensitive change so let's run core perf tests before merging this PR.